azure-pipelines.yml

```diff
@@ -242,53 +241,48 @@ jobs:
       versionSpec: '$(python.version)'
 
   - bash: |
+      set -e
       sed -i "s/'giotto-tda'/'giotto-tda-nightly'/1" setup.py
       sed -i "s/__version__.*/__version__ = '$(Build.BuildNumber)'/1" gtda/_version.py
-    failOnStderr: true
     condition: eq(variables['nightly_check'], 'true')
     displayName: 'Change name to giotto-tda-nightly'
 
   - script: |
       python -m pip install --upgrade pip setuptools
       pip install -e ".[tests, doc]"
-    failOnStderr: true
     displayName: 'Install dev environment'
 
   - script: |
-      pytest gtda --cov --cov-report xml
+      pytest gtda --cov --cov-report xml || exit /b
       flake8
-    failOnStderr: true
     displayName: 'Test dev install with pytest and flake8'
 
   - script: |
-      pip uninstall -y giotto-tda || exit /b
-      pip uninstall -y giotto-tda-nightly || exit /b
+      pip uninstall -y giotto-tda
+      pip uninstall -y giotto-tda-nightly
```

Comment:
Honestly I don't think that checking for exit status here is that useful, when running pip install previous versions will be uninstalled anyway and we should never have both gtda and gtda-nightly simultaneously in CI.

Reply:
Ok no problem!